### PR TITLE
fix(form-v2): implement full list numbering for TOU and PP pages

### PR DESCRIPTION
**Pull request by karrui**
> This PR is built on  https://github.com/opengovsg/FormSG/pull/4259, and supersedes that PR.

## Problem
Numbering in the react version's terms of use page and privacy policy page is inconsistent with angular JS app as outlined in #4111 . 

Closes [#4111 ]

## Solution
This PR resolves this issue by implementing a customizable reusable component which is able to perform nested listing with the correct sequence marker syntax intended. Refer to the following links for a more comprehensive understanding
https://developer.mozilla.org/en-US/docs/Web/CSS/counter 
https://developer.mozilla.org/en-US/docs/Web/CSS/counters 
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  

**Features**:

- refer to pictures

**Improvements**:
- terms of use page and policy page had components which were basically clones of each other. I refactored them into reusable components and removed the redundant code.

## Before & After Screenshots

**BEFORE**:
- refer to #4111 there are a few pictures there

**AFTER**:

<details>
<summary>terms of use page</summary>

terms_of_use_page_after
</details>

<details>
<summary>terms of use page with css borders</summary>

terms_of_use_page_after_with_css
</details>

<details>
<summary>privacy page </summary>

privacy_page_after
</details>

<details>
<summary>privacy page with css borders</summary>

privacy_page_after_with_css

</details>

## Tests

- terms of use and policy don't need any unit test as they are purely cosmetic

### CODE REVIEW

**Review by wanlingt — APPROVED** [review]
lgtm!

**Review by timotheeg — APPROVED** [review]

File: frontend/src/templates/ResendOtpButton/ResendOtpButton.stories.tsx [hunk 1]
```diff
@@ -26,8 +29,9 @@ InProgress.args = {
   isDisabled: true,
 }
 
-export const Playground = () => (
-  <ResendOtpButtonContainer
-    onResendOtp={() => new Promise((res) => setTimeout(res, 800))}
-  />
+export const Playground: Story<ResendOtpButtonContainerProps> = (args) => (
```
Comment (mantariksh) [inline frontend/src/templates/ResendOtpButton/ResendOtpButton.stories.tsx:32]: ???

**Review by mantariksh — APPROVED** [review]
gtlm

File: frontend/src/templates/ResendOtpButton/ResendOtpButton.stories.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/templates/ResendOtpButton/ResendOtpButton.stories.tsx:32]: lmao rubbish from trying to solve another PR LOL. but at least this makes it easier to use that story :P

**Pull request merged**